### PR TITLE
Validates uniqueness of the adviser reference number.

### PR DESCRIPTION
**Pull request by lshepstone**
@benlovell

### CODE REVIEW

**Comment by lshepstone** [comment]
@alexwllms just need to consume the returned error message now

File: spec/models/adviser_spec.rb [hunk 1]
```diff
@@ -87,6 +87,18 @@
           expect(a).to_not be_valid
         end
       end
+
+      context 'when an adviser with the same reference number already exists' do
+        let(:reference_number) { 'ABC12345' }
+
+        before do
+          create(:adviser, reference_number: 'ABC12345')
```
Comment (benlovell) [inline spec/models/adviser_spec.rb]: You didn't use your `let`'d `reference_number` here.

**Comment by benlovell** [comment]
:+1: :ship: :shipit:

**Comment by jongilbraith** [comment]
:+1:

**Pull request merged**